src/main/java/com/hartwig/platinum/config/SampleBucket.java

```diff
@@ -0,0 +1,58 @@
+package com.hartwig.platinum.config;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+
+public class SampleBucket {
+
+    private static final String TUMOR_PATH = "TUMOR";
+    private static final String NORMAL_PATH = "NORMAL";
+    public static final String R1 = "R1";
+    public static final String R2 = "R2";
+    private final Bucket bucket;
+
+    public SampleBucket(final Bucket bucket) {
+        this.bucket = bucket;
+    }
+
+    public List<SampleConfiguration> apply() {
+        return StreamSupport.stream(bucket.list(Storage.BlobListOption.currentDirectory()).iterateAll().spliterator(), false)
+                .map(blob -> ImmutableSampleConfiguration.builder()
+                        .addTumors(extractFastq(blob, TUMOR_PATH))
+                        .normal(extractFastq(blob, NORMAL_PATH))
+                        .name(blob.getName().replace("/", ""))
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private RawDataConfiguration extractFastq(final Blob blob, final String typePrefix) {
+        final String sampleName =
+                StreamSupport.stream(bucket.list(Storage.BlobListOption.prefix(blob.getName() + typePrefix)).iterateAll().spliterator(),
+                        false).findFirst().map(BlobInfo::getName).map(s -> s.split("/")).map(s -> s[2]).orElseThrow();
+        final Iterable<Blob> fastqs = bucket.list(Storage.BlobListOption.prefix(blob.getName() + typePrefix)).iterateAll();
+        ImmutableRawDataConfiguration.Builder rawDataConfigurationBuilder = ImmutableRawDataConfiguration.builder();
+        for (Blob fastq : fastqs) {
+            if (fastq.getName().contains(R1)) {
```

Comment:
Are these strings specific enough? Are we really never going to encounter the string `R1` or `R2` anywhere other than the end of the filename? Just wondering if an anchor would be a good idea (eg in light of the new naming for second samples).

Reply:
for now i think its fine.